serializers/src/main/java/io/pravega/schemaregistry/serializers/MultiFormatSerializerFactory.java

```diff
@@ -0,0 +1,199 @@
+        autoCreateGroup(schemaRegistryClient, config);
+        registerCodec(schemaRegistryClient, config);
+        String groupId = config.getGroupId();
+
+        // if serializer is not already present, create a new serializer. 
+        Function<SchemaInfo, AbstractPravegaSerializer<Object>> serializerFunction =
+                x -> getPravegaSerializer(config, customSerializers, schemaRegistryClient, groupId, x);
+        return new MultipleFormatSerializer(serializerFunction);
+    }
+
+    private static <T> Serializer<T> deserializeAsTInternal(SerializerConfig config,
+                                                    Map<SerializationFormat, PravegaDeserializer<Object>> deserializers,
+                                                    BiFunction<SerializationFormat, Object, T> transform) {
+        String groupId = config.getGroupId();
+        SchemaRegistryClient schemaRegistryClient = config.getRegistryConfigOrClient().isLeft() ?
+                SchemaRegistryClientFactory.createRegistryClient(config.getRegistryConfigOrClient().getLeft()) :
+                config.getRegistryConfigOrClient().getRight();
+        autoCreateGroup(schemaRegistryClient, config);
+        failOnCodecMismatch(schemaRegistryClient, config);
+        EncodingCache encodingCache = new EncodingCache(groupId, schemaRegistryClient);
+
+        AbstractPravegaDeserializer json = new JsonGenericDeserlizer(config.getGroupId(), schemaRegistryClient,
+                config.getDecoder(), encodingCache);
+        AbstractPravegaDeserializer protobuf = new ProtobufGenericDeserlizer(groupId, schemaRegistryClient, null, config.getDecoder(),
+                encodingCache);
+        AbstractPravegaDeserializer avro = new AvroGenericDeserlizer(groupId, schemaRegistryClient, null, config.getDecoder(),
+                encodingCache);
+
+        Map<SerializationFormat, AbstractPravegaDeserializer> map = new HashMap<>();
+        map.put(SerializationFormat.Json, json);
+        map.put(SerializationFormat.Avro, avro);
+        map.put(SerializationFormat.Protobuf, protobuf);
+
+        deserializers.forEach((key, value) -> {
+            map.put(key, new AbstractPravegaDeserializer<Object>(groupId, schemaRegistryClient, null, false, config.getDecoder(), encodingCache) {
+                @Override
+                protected Object deserialize(InputStream inputStream, SchemaInfo writerSchema, SchemaInfo readerSchema) {
+                    return value.deserialize(inputStream, writerSchema, readerSchema);
+                }
+            });
+        });
+
+        return new MultipleFormatDeserializer<>(groupId, schemaRegistryClient, map, config.getDecoder(),
+                encodingCache, transform);
+    }
+
+    private static <T> Serializer<WithSchema<T>> deserializerInternal(SerializerConfig config, Map<SerializationFormat,
+            PravegaDeserializer<Object>> deserializers, BiFunction<SerializationFormat, Object, T> transform) {
```

Comment:
And here

Reply:
https://github.com/pravega/schema-registry/pull/41#discussion_r449404120